src/main/kotlin/app/revanced/patches/all/connectivity/wifi/spoof/patch/SpoofWifiPatch.kt

```diff
@@ -0,0 +1,232 @@
+            arrayOf("Landroid/net/NetworkRequest;", "Landroid/net/ConnectivityManager\$NetworkCallback;"),
+            "V",
+        ),
+        RequestNetwork2(
+            "Landroid/net/ConnectivityManager;",
+            "requestNetwork",
+            arrayOf("Landroid/net/NetworkRequest;", "Landroid/net/ConnectivityManager\$NetworkCallback;", "I"),
+            "V",
+        ),
+        RequestNetwork3(
+            "Landroid/net/ConnectivityManager;",
+            "requestNetwork",
+            arrayOf("Landroid/net/NetworkRequest;", "Landroid/net/ConnectivityManager\$NetworkCallback;", "Landroid/os/Handler;"),
+            "V",
+        ),
+        RequestNetwork4(
+            "Landroid/net/ConnectivityManager;",
+            "requestNetwork",
+            arrayOf("Landroid/net/NetworkRequest;", "Landroid/app/PendingIntent;"),
+            "V",
+        ),
+        RequestNetwork5(
+            "Landroid/net/ConnectivityManager;",
+            "requestNetwork",
+            arrayOf("Landroid/net/NetworkRequest;", "Landroid/net/ConnectivityManager\$NetworkCallback;", "Landroid/os/Handler;", "I"),
+            "V",
+        );
+
+        fun replaceInstruction(method: MutableMethod, instruction: Instruction35c, instructionIndex: Int) {
+            val registers = arrayOf(instruction.registerC, instruction.registerD, instruction.registerE, instruction.registerF, instruction.registerG)
+            val argsNum = this.methodParams.size + 1 // + 1 for instance of definedClassName
+            if (argsNum > registers.size) {
+                // should never happen, but just to be sure (also for the future) a safety check
+                throw RuntimeException("not enough registers for ${this.definedClassName}#${this.methodName}: required $argsNum registers, but only got ${registers.size}")
+            }
+            val args = registers.take(argsNum).joinToString(separator = ", ") { reg -> "v${reg}" }
+            val replacementMethodDefinition = "${this.methodName}(${this.definedClassName}${this.methodParams.joinToString(separator = "")})${this.returnType}"
+            method.replaceInstruction(
+                instructionIndex,
+                "invoke-static { $args }, $INTEGRATIONS_CLASS_DESCRIPTOR->${replacementMethodDefinition}"
+            )
+        }
+
+        companion object {
+            fun fromMethodReference(methodReference: MethodReference) = values().firstOrNull { search ->
+                search.definedClassName == methodReference.definingClass
+                        && search.methodName == methodReference.name
+                        && methodReference.parameterTypes.toTypedArray().contentEquals(search.methodParams)
+            }
+        }
+    }
+
+    override fun filterMap(
+        classDef: ClassDef,
+        method: Method,
+        instruction: Instruction,
+        instructionIndex: Int
+    ): InstructionInfo? {
+        if (classDef.type.startsWith(INTEGRATIONS_CLASS_DESCRIPTOR.removeSuffix(";"))) {
```

Comment:
Wouldn't it be sufficient to check for equality since we can assume there is only one class that has to be considered to avoid recursion? This would allow getting rid of calling `removeSuffix` for each instruction in all classes and methods.

Reply:
Yes, it would be sufficient. Although it isn’t stated in the comment, another reason to do this check is to simply avoid messing up the logic of our patch, so that we can still see what the real network info looks like and so on. Currently, there is no such logic in the subclasses, but I’d still prefer to keep it that way as that might change in the future and it keeps the instruction replacement logic consistent for the integrations class and its subclasses. Also, I don’t think the `removeSuffix` call has a huge performance impact.